File: phoenix/lib/cards_project/content/draft_participant.ex
defmodule CardsProject.Content.DraftParticipant do
  use Ecto.Schema
  import Ecto.Changeset

  schema "draft_participants" do
    field :seat_number, :integer
    field :joined_at, :naive_datetime
    belongs_to :session, CardsProject.Content.DraftSession
    belongs_to :player, CardsProject.Players.Player
    belongs_to :drafted_cards, CardsProject.Content.DraftPick

    timestamps()
  end

  @doc false
  def changeset(record, attrs) do
    record
    |> cast(attrs, [:seat_number, :joined_at, :session_id, :player_id])
    |> validate_required([:seat_number, :joined_at])
  end

  # ── Business operations ────────────────────────────────────────────

  def pick_card(_record, _card_id, _pack_number) do
    # TODO: implement DraftParticipant.pick_card
    :ok
  end

  def drafted_card_count(_record) do
    # TODO: implement DraftParticipant.drafted_card_count
    {:error, :not_implemented}
  end
end

File: laravel/app/Models/Content/DraftParticipant.php
<?php

namespace App\Models\Content;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use App\Models\Players\Player;

class DraftParticipant extends Model
{
    protected $table = 'draft_participants';

    protected $fillable = ['seat_number', 'joined_at', 'session_id', 'player_id'];

    protected $casts = [
        'joined_at' => 'datetime',
    ];

    public function session(): BelongsTo
    {
        return $this->belongsTo(DraftSession::class, 'session_id');
    }

    public function player(): BelongsTo
    {
        return $this->belongsTo(Player::class, 'player_id');
    }

    // ── Validation rules ─────────────────────────────────────────────
    public function validateRules(): void
    {
        $errors = [];
        if (!(($this->seat_number === null || $this->seat_number > 0))) {
            $errors['seat_number_positive'] = 'Seat number must be greater than zero';
        }
        if (!empty($errors)) {
            throw new \Illuminate\Validation\ValidationException(
                \Illuminate\Support\Facades\Validator::make([], []),
                response()->json(['errors' => $errors], 422)
            );
        }
    }

    // ── Business operations ──────────────────────────────────────────

    public function pickCard($card_id, $pack_number): void
    {
        throw new \RuntimeException('pick_card not implemented');
    }

    public function draftedCardCount(): int
    {
        throw new \RuntimeException('drafted_card_count not implemented');
    }

}
